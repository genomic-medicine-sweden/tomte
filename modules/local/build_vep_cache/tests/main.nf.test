nextflow_process {

    name "Test Process BUILD_VEP_CACHE"
    script "../main.nf"
    process "BUILD_VEP_CACHE"
    tag "modules"
    tag "modules_local"
    tag "build_vep_cache"

    setup {
        run("VEP_GNOMAD_DOWNLOAD") {
            script "../../download/vep_gnomad_download/main.nf"
            process {
                """
                input[0] = 'GRCh38'
                input[1] = 110
                """
            }
        }
        run("WGET_DOWNLOAD") {
            script "../../download/wget_download/main.nf"
            options "-stub"
            process {
                """
                input[0] = Channel.of([
                    'vep_cache_and_plugins',
                    "https://raw.githubusercontent.com/nf-core/test-datasets/raredisease/reference/vep_cache_and_plugins.tar.gz",
                ])
                """
            }
        }
    } 

    
    test("Should run without failures") {

        when {
            options "-stub VEP_GNOMAD_DOWNLOAD"
            process {
                """
                input[0] = WGET_DOWNLOAD.out.downloaded_file.collect()
                input[1] = VEP_GNOMAD_DOWNLOAD.out.gnomad_vcf_tbi.flatten().collect()
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
