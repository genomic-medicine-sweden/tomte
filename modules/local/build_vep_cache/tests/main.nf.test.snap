{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        
                    ]
                ],
                "1": [
                    "vep_plugin_files.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    "versions.yml:md5,c63c2a035f67c62ce2cc7a732c32d355"
                ],
                "vep_cache": [
                    [
                        
                    ]
                ],
                "vep_plugin_file": [
                    "vep_plugin_files.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,c63c2a035f67c62ce2cc7a732c32d355"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-23T12:00:00.200818"
    }
}